src/store/createMemoryOutstandingStore.ts

```diff
@@ -0,0 +1,312 @@
+                    isVersion06(mempoolUserOp) &&
+                    userOp.initCode &&
+                    userOp.initCode !== "0x" &&
+                    mempoolUserOp.initCode &&
+                    mempoolUserOp.initCode !== "0x" &&
+                    isSameSender
+
+                const isConflictingV7Deployment =
+                    isVersion07(userOp) &&
+                    isVersion07(mempoolUserOp) &&
+                    userOp.factory &&
+                    userOp.factory !== "0x" &&
+                    mempoolUserOp.factory &&
+                    mempoolUserOp.factory !== "0x" &&
+                    isSameSender
+
+                const isConflictingDeployment =
+                    isConflictingV6Deployment || isConflictingV7Deployment
+
+                if (isConflictingDeployment) {
+                    conflictingReason = {
+                        reason: "conflicting_deployment",
+                        userOp: userOpInfo.userOp
+                    }
+                    break
+                }
+            }
+
+            return conflictingReason
+        },
+        contains: async (userOpHash: HexData32) => {
+            for (const userOpInfos of pendingOps.values()) {
+                if (
+                    userOpInfos.some((info) => info.userOpHash === userOpHash)
+                ) {
+                    return true
+                }
+            }
+            return false
+        },
+        peek: () => {
+            if (priorityQueue.length === 0) {
+                return Promise.resolve(undefined)
+            }
+
+            return Promise.resolve(priorityQueue[0])
+        },
+        pop: () => {
+            const userOpInfo = priorityQueue.shift()
+
+            if (!userOpInfo) {
+                return Promise.resolve(undefined)
+            }
+
+            const pendingOpsSlot = senderNonceSlot(userOpInfo.userOp)
+            const backlogOps = pendingOps.get(pendingOpsSlot)
+
+            // This should never throw.
+            if (!backlogOps?.shift()) {
+                throw new Error("FATAL: No pending userOps for sender")
+            }
+
+            // Move next pending userOp into priorityQueue if exist.
+            if (backlogOps.length > 0) {
+                addToPriorityQueue({ userOpInfo: backlogOps[0], priorityQueue })
+            }
+
+            // Cleanup.
+            if (backlogOps.length === 0) {
+                pendingOps.delete(pendingOpsSlot)
+            }
+
+            return Promise.resolve(userOpInfo)
+        },
+        add: (userOpInfo: UserOpInfo) => {
+            const { userOp, userOpHash } = userOpInfo
+            const [nonceKey] = getNonceKeyAndSequence(userOp.nonce)
+            const pendingOpsSlot = senderNonceSlot(userOp)
+
+            const backlogOps =
+                pendingOps.get(pendingOpsSlot) ||
+                pendingOps.set(pendingOpsSlot, []).get(pendingOpsSlot)!
+
+            backlogOps.push(userOpInfo)
```

Comment:
can we leave a comment that this is a push to the reference of the array so this will update the state in pendingOps